docs/pages/deploy-a-cluster.mdx

```diff
@@ -0,0 +1,32 @@
+---
+title: Deploying a Self-Hosted Teleport Cluster
+description: Guides for deploying self-hosted Teleport clusters on various platforms and cloud providers.
+---
+
+{/*TOPICS*/}
```

Comment:
What does this do? DO we have a reference of all the magic syntax we can use in our doc engine?

Reply:
I added a table of contents page generator to the feature branch I'm using for docs reorganization (it's not in `master`). This ensures that we have a comprehensive set of TOC pages as I move docs around. Here's the README: https://github.com/gravitational/teleport/tree/paul.gottschling/2024-06-reorg/docs/gen-topic-pages

Reply:
I'll be more explicit that these PRs are for my feature branch, not `master`